x/crosschain/keeper/evm_deposit.go

```diff
@@ -48,36 +51,41 @@ func (k msgServer) HandleEVMDeposit(ctx sdk.Context, cctx *types.CrossChainTx, m
 		evmTxResponse, err := k.fungibleKeeper.ZRC20DepositAndCallContract(ctx, from, to, msg.Amount.BigInt(), senderChain, msg.Message, contract, data, msg.CoinType, msg.Asset)
 		if err != nil {
 			isContractReverted := false
-			if evmTxResponse != nil && evmTxResponse.Failed() {
+
+			// consider the contract as reverted if foreign coin liquidity cap is reached
+			if (evmTxResponse != nil && evmTxResponse.Failed()) || errors.Is(err, fungibletypes.ErrForeignCoinCapReached) {
 				isContractReverted = true
 			}
+
 			return isContractReverted, err
 		}
 
 		// non-empty msg.Message means this is a contract call; therefore the logs should be processed.
 		// a withdrawal event in the logs could generate cctxs for outbound transactions.
 		if !evmTxResponse.Failed() && len(msg.Message) > 0 {
 			logs := evmtypes.LogsToEthereum(evmTxResponse.Logs)
-			ctx = ctx.WithValue("inCctxIndex", cctx.Index)
-			txOrigin := msg.TxOrigin
-			if txOrigin == "" {
-				txOrigin = msg.Sender
-			}
+			if len(logs) > 0 {
```

Comment:
Does `len(logs) == 0` mean something wrong (e.g., we always expect some logs)? Just curious about what pre-assumption we had in our mind.

Reply:
It prevents doing unnecessary call to fetch system contracts since there would be no logs to iterate